includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -795,11 +795,13 @@ private function process_payment_with_deferred_intent( int $order_id ) {
 				$this->update_saved_payment_method( $payment_method_id, $order );
 			}
 
+			// Lock the order before we create and confirm the payment/setup intents to prevent Stripe sending the success webhook before this request is completed.
+			$this->lock_order_payment( $order );
```

Comment:
Hey @Mayisha, in the changes you made in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3331, the lock was added after the `validate_minimum_order_amount()` call and so I've been trying to test and break this PR to see if there's any issues with it being before `validate_minimum_order_amount()`.

When I tested with a $0.40 product, the lock gets added even though there's a checkout error, but I haven't been able to break it because when I add 2 of these to meet the minimum amount, the checkout is then successful.

I wanted to ping you on this issue to ask if you discovered any issues with this?

The alternative was to either add two calls to `lock_order_payment()` for both payment/setup intent calls or move the `validate_minimum_order_amount()` higher up in the `process_payment_with_deferred_intent()` function 🤔 
Let me know if you had any thoughts :)

Reply:
You are right Matt. I intended to add the lock right before processing the intent if it passes `validate_minimum_order_amount`. 
However, I also could not break it and the checkout was successful on second attempt (when I increased the number of items or add a shipping charge).

What happens here is, on the first attempt an order is created and the order status is `failed`. On the second attempt a new order gets created then successfully gets processed and so the lock on the first order does not hamper the checkout.

<img width="1286" alt="Screenshot 2024-09-06 at 12 20 37 AM" src="https://github.com/user-attachments/assets/f3368f6d-1b13-4187-9502-f703f2fbcf44">

In the database, the transient of the first failed orders remain.

<img width="828" alt="Screenshot 2024-09-06 at 1 23 29 AM" src="https://github.com/user-attachments/assets/70d8b7b4-ae9d-4273-9168-f67b62aec6f1">

It is alright in my opinion as the transient will be cleared once it reaches the expiration time.